src/main/scala/io/radicalbit/nsdb/connector/kafka/sink/conf/MappingConfUtility.scala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2019 Radicalbit S.r.l.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.radicalbit.nsdb.connector.kafka.sink.conf
+
+import com.datamountaineer.kcql.Kcql
+import io.radicalbit.nsdb.connector.kafka.sink.NSDbSinkWriter.validateDefaultValue
+import io.radicalbit.nsdb.connector.kafka.sink.conf.Constants.{KcqlType, MapType, MappingType}
+import io.radicalbit.nsdb.connector.kafka.sink.models.{MappingInterface, ParsedKcql, EnrichedMapping, Mappings}
+import org.apache.kafka.common.config.ConfigValue
+
+import scala.util.Try
+
+object SupportStructure {
+  val empty: SupportStructure =
+    SupportStructure(List.empty[String], List.empty[String], List.empty[String], List.empty[String])
+}
+
+final case class SupportStructure(metrics: List[String],
+                                  values: List[String],
+                                  timestamps: List[String],
+                                  tags: List[String])
+
+trait MappingConfUtility {
+  import io.circe.generic.auto._
+  import io.circe.parser._
+  import io.circe.syntax._
+
+  // consider that at this point each fields format is validated
+  private def parseMappings(metrics: String,
+                            values: Option[String],
+                            timestamps: Option[String],
+                            tags: Option[String]) = {
+
+    val metricsFieldName    = "METRICS"
+    val valuesFieldName     = "VALUES"
+    val timestampsFieldName = "TIMESTAMPS"
+    val tagsFieldName       = "TAGS"
+
+    val dottedNotation = "[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+".r
```

Comment:
here we can rename into dottedNotationToken or similar. Up to you, but I would avoid any naming overlapping